BROWSER_README.md

```diff
@@ -0,0 +1,99 @@
+    provide an evergreen link that does not expire.
+
+## The problem: Display data from an API for human consumption
+
+Please write a static webpage that loads data from `https://swapi.dev/api/planets/` and displays that data
+in an HTML `table`.
+
+Name the entrypoint for your application `index.html`.
+
+You may use additional files to organize your code. You may also use frameworks and build tools as long as clear, simple instructions are provided for how to run. Please ensure your solution can be easily viewed in the browser. Be ready to answer questions about why you chose a given approach.
+
+Don't worry about making the table look fancy; you can style it however you
+would like to outside the requirements below. We will look at the markup you
+generate and any code you write to load or display the data.
+
+### What to display
+
+For each planet in the dataset, please display:
+
+-   The planet's name
+    -   The name should be a link that, when clicked, opens the planet's API URL in a new window
+-   The planet's climate
+-   How many residents the planet has
+-   The terrains found on the planet
+-   The population
+-   The surface area covered by water
+    -   Assume that all planets are perfect spheres.
+    -   The radius of a sphere is half its diameter.
+    -   The value of `surface_water` from the API is a percentage, so a value of `50` means the planet is 50% covered in water.
+    -   Round these values to the nearest integer value.
+
+Additionally, please satisfy these requirements:
+
+-   Show a loading message while loading the data, and hide this message once the data is displayed.
+-   Display an error message if the data load fails for some reason. We may test your code against an invalid URL to demonstrate this.
+-   Sort the planets alphabetically
+-   Any column that is unknown should be displayed as `?`.
+-   Format all numbers larger than 999 with spaces to group digits into groups of three.
+    For example, ten thousand should be displayed as `10 000`.
+-   Cells in the table should be separated by 1 pixel gray lines.
+
+### Stretch goals (optional)
+
+If you complete the above and still have additional time, you can choose to complete one or more of the following extra requirements.
+
+**This is absolutely not required for submitting the work sample and should not be worked on beyond the four hour time limit.**
+
+-   The API only returns the first page of data by default. Add a "load more" button that
+    loads additional pages of data when clicked. Only show this button if there is more data to be loaded.
+-   Sort the table by a column's values when that column's header is clicked.
+-   Table has both an empty state and loading state (when no data exists and when data is still loading accordingly).
+-   Spruce up your page with some extra styling & design elements!
+-   Make sure your page is usable on mobile devices.
+-   Include some tests.
```

Comment:
Making this a stretch goal might actually discourage someone from writing tests although they would have written tests had they never been asked. It is useful to see if an engineer writes tests even when it's not explicitly required (or explicitly optional, for that matter).

Reply:
the reason for making this a stretch goal (in my opinion) is that there is no native test runner for browser code, so unless you have already decided to use a JS framework, there will be a not insignificant amount of setup to get tests running.